### ISSUE TITLE
Some sentences are correct but shown in red

### ISSUE THREAD

**Issue by trang** [Tatoeba/tatoeba2#1659]
## Problem

As mentioneed in https://github.com/Tatoeba/tatoeba2/pull/1650#issuecomment-417822863
> We have many sentences that are marked as "unapproved" only because of possible copyright issues. Having them displayed in red with a warning sign may give the wrong idea to regular users, who'd think there's something incorrect in the sentence, but there isn't.
>
> The reason why they're marked as "unapproved" is because unapproved sentences are excluded from the CSV exports. This is a workaround rather than an actual solution. 

Some of these sentences can be found by looking at sentences that have a tag containing "copyright":
https://tatoeba.org/eng/tags/view_all/copyright

## Suggested solution

1. Allow admins to change the license from CC-BY to `null`. Not sure what label to put for license `null` though. No license? Unknown license? Tatoeba only?
2. Exclude sentences that are not CC-BY from the CSV exports.
3. Prevent users from translating these sentences, at least until we figure out how to handle the license of translations.
4. Display the license info to everyone, not just to admins (we can then close #1555).

**Comment by jiru** [comment]
I think it’s a good solution, because people looking for the license of the sentence should be warned about licensing issues, and because it’s simple.

> Allow admins to change the license from CC-BY to null. Not sure what label to put for license null though. No license? Unknown license? Tatoeba only?

How about "Licensing issue"? Another question: any particular reason for using `null` instead of a regular value?

**Comment by trang** [comment]
> How about "Licensing issue"?

That works too.

> Another question: any particular reason for using null instead of a regular value?

No particular reason. I guess it just feels more accurate to use `null` because to me regular values to should refer to actual licenses, so I naturally fallback to `null` whenever to express "stuff that's not defined".

**Comment by sacredceltic** [comment]
I can't see why users should be prevented from translating these sentences. The license issue is with the sentence itself, not its potential translations, which could well be original and genuine...

**Comment by jiru** [comment]
@sacredceltic A translation is a derivative work. As such, if the license of the original work doesn’t allow publishing a translation under CC-BY compatible terms, one cannot legally add a translation of it on Tatoeba.

For example, let’s say a given sentence is part of the lyrics of a song. It’s not legal to publish any unauthorized translation of the lyrics, because the song author owns the exclusive right to make derivatives from his work. So not only the sentence, as part of lyrics, cannot be published on Tatoeba because the corpus is licensed under CC-BY, but also its translations are infringing the copyright of the original lyrics.

**Comment by AndiPersti** [comment]
@Trang @jiru 
I'm on vacation the next week so I should have enough time to work on this issue (at first sight it doesn't look too complicated).

> Allow admins to change the license from CC-BY to null. Not sure what label to put for license null though. No license? Unknown license? Tatoeba only?

I have some sentences in my local database which have `NULL` as their license because I've manually added them from the console. These are shown in the UI as `for Tatoeba only`.

1) Are there sentences in the production database with license `NULL`?
2) Do we want to distinguish between such sentences and sentences with copyright issues? If yes, we can't use `NULL` for the latter ones.

**Comment by trang** [comment]
Thanks for looking into this @AndiPersti 🙂 

There are [sentences with a null license](https://gist.github.com/trang/b692ad34ed198e819bc270e52d493c12) but I'm not sure under which circumstance this happened. I don't think there was an intention to have the license of these sentences set to null. I cannot think of  a use case for this. It could be the result of a bug or it could be something that was meant to be temporary but we forgot about it.

The label `for Tatoeba only` was introduced for audio initially. When we started recording audio for our sentences, we did not think much about their license. When we realized many people would love to reuse Tatoeba's audio, we introduced licenses on audio but we could not get back in touch with all the contributors to ask them which license they would like to use. So some audio remained with no license (hence "for Tatoeba only").

For sentences, we never did something like that. Technically, there are some sentences from the pre-CC-BY era which, strictly speaking, should be under `for Tatoeba only`. But (as far as I remember) we never invested effort into reviewing those sentences and set their license to `null` when appropriate. So in theory there shouldn't be any sentence with a `null` license.

Regarding the wording:
- We should definitely not label copied non CC BY compatible sentences as `for Tatoeba only`, that would be wrong. For these sentences, `Licensing issues` would be more appropriate.
- But for our audio, `Licensing issues` would feel incorrect and `for Tatoeba only` is more appropriate.

I think `for Tatoeba only` is a bit misleading though, because it feels like the contributor's wish was explicitly that Tatoeba does not share their audio, whereas in fact, we don't know what is their wish.

And to be fair, for non CC BY compatible sentences we technically don't know the original's author wish either. So I'd be tempted to try and come up with a wording that convey this idea for both textual and audio contributions.

**Comment by agrodet** [comment]
May I ask you to make sure to give a lot of thought before dealing with this issue and make sure that the solution isn't worse than the problem?

  After a night of thoughts, I realized I had a big problem with this. Most of what follows is a personal   opinion.    
I want to make sure that we are not talking about a new policy: "adding copyrighted contents to Tatoeba is now okay!" because I think that's a terrible thing.

First of all, we will not have any countermeasure against people who will scrap other websites and add their contents to Tatoeba. Someone could simply take all sentences of a website, a song, a movie, or a book, and add them to Tatoeba. "That's okay because it's copyrighted but we won't distribute it." 
What is the purpose of Tatoeba? I thought we were not working for individuals, not even for Tatoeba users, but for everybody who need a good corpus of sentences. This appears to me as a blow against the power of sharing that Tatoeba represent. 
Copyrighted content, if allowed, shouldn't be treated the same way as the usual contributions. I don't want to have anything to do with copyrighted content. I want to share my knowledge with the rest of the world. I don't care about capricious pricks that don't have inspiration and only find pleasure in sabotaging other people's work. I don't want to see copyrighted content appear when I search, when I translate, I don't want it to make me loose my time when I proofread the French corpus, and in the best of worlds, I don't want other people spending efforts on something that cannot be shared. 
By allowing copyrighted content, we're adding a wall to access Tatoeba's content. Not a paywall, like others do sometimes, but you're saying that if I don't register, I cannot access a part of your content. What kind of vision is that.

Actually, everything I describe is taken care of pretty well by the red marking. Potential copyright infringement could, or should, be labeled "potential copyright infringement" to explain why a sentence is red. Other problematic sentences should be tagged with the reason. I can't see any reason to negotiate with people who can't read or respect a rule that is clearly stated, and even less reasons on why we should sabotage ourselves for them.
**I think that if there is something to improve, it is the red marking.** Any red sentence should have a tag explaining the reason. That way, users would know the reason and could potentially discuss it, and it would also gives indication to new users about the kind of content that cannot be added to Tatoeba. Going back to the original statement of the problem 

>  We have many sentences that are marked as "unapproved" only because of possible copyright issues. Having them displayed in red with a warning sign may give the wrong idea to regular users, who'd think there's something incorrect in the sentence, but there isn't.
    The reason why they're marked as "unapproved" is because unapproved sentences are excluded from the CSV exports. This is a workaround rather than an actual solution.

the improved red warning would not let people think there is something incorrect in the sentence, but clearly let them know that this content is inappropriate for Tatoeba, and with a tag (or a better mean), they would know why. Therefore, even after implementing your proposed solution, those sentences should be marked in red (so actually, I don't see the problem).

**Comment by ckjpn** [comment]
> Actually, everything I describe is taken care of pretty well by the red marking.

Note that any time content is on the web, it's "being distributed," so I'm not sure marking things in red really takes care of the problem.  It does, perhaps, solve the problem for re-users, but not for tatoeba.org itself.  Perhaps even showing logs of deleted copyrighted material is not a good idea, since people could possibly still be seeing these sentences on tatoeba.org.

**Comment by trang** [comment]
> I want to make sure that we are not talking about a new policy: "adding copyrighted contents to Tatoeba is now okay!"

We are not. To be clear on this: copyrighted content is not allowed and will never be allowed. If we find copyrighted content and we have no doubt about it, we delete it.

Our guidelines about avoiding copy-pasting will remain and we will continue to promote the respect of other people's work.

But our problem is about dealing with content for which the copyright status is unclear. The notion of intellectual property has been quite shaken up with the rise of the Internet and there are some cases where it's hard to draw the line. Especially when we are not specialists on this topic.

> First of all, we will not have any countermeasure against people who will scrap other websites and add their contents to Tatoeba.

We have countermeasures. We can block the user and delete all their sentences.

> I think that if there is something to improve, it is the red marking. Any red sentence should have a tag explaining the reason. That way, users would know the reason and could potentially discuss it, and it would also gives indication to new users about the kind of content that cannot be added to Tatoeba.

I didn't think much about having a specific visual indicator for sentences with copyright issues, but it's a fair point. 

The only argument I'd have against this idea is that Tatoeba's focus is on the linguistic aspect. If we're going to squeeze more indicators into the sentence component, it should be motivated by linguistic reasons primarily. So if a sentence is linguistically questionable, then yes, I think it's quite straightforward to have an indicator for it.

But for other reasons, I'm a bit more hesitant. It doesn't mean we shouldn't mention the problem at all (it can be mentioned on the sentence's page). But to mention it on the sentence's component itself feels like putting too much burden on it. If I'm an outsider and will just be viewing these sentences through the search results, what would an indicator on copyright issues bring to me?

> Therefore, even after implementing your proposed solution, those sentences should be marked in red (so actually, I don't see the problem).

Assuming we decide it's worth having an indicator for copyright issues, we would have to figure out different visuals.

I would be okay, as a short term solution, that we only change the tooltip text to something like `This sentence has copyright issues` (instead of `This sentence is not reliable`). In the long term though, I think those two issues should be visually different enough, so that the user isn't tricked to think that a sentence has mistakes when it only has a copyright issue or vice versa. 

More concretely, the kind of scenario I want to avoid is:
- I'm a new user, I stumble upon a sentence that is red.
- I hover over the warning icon and it says "This sentence is unreliable".
- Some time later, I see another red sentence. I don't check the tooltip, I automatically assume it has something wrong, linguistically speaking.
- But it turns out it's a perfectly correct sentence. It just has copyright issues.

**Comment by AndiPersti** [comment]
> There are [sentences with a null license](https://gist.github.com/trang/b692ad34ed198e819bc270e52d493c12) but I'm not sure under which circumstance this happened. I don't think there was an intention to have the license of these sentences set to null. I cannot think of a use case for this. It could be the result of a bug or it could be something that was meant to be temporary but we forgot about it.

After a quick look at a few random sentences on that list I've noticed that most of them are either orphan or the current owner is different from the original one. So maybe something to investigate further.

**Comment by AndiPersti** [comment]
Following the discussion here and exploring some sentences tagged with copyright issues(*), I wonder whether the suggested solution is enough.

Suppose we implement the changes. What will happen afterwards?
The problematic sentence is excluded from the export and can't be translated. But it is still in the database and will be displayed either by going to the Tags page or including unapproved sentences in the search.
So I agree with CK and I think the sentence shouldn't be displayed at all anywhere on tatoeba.org except perhaps a special page for admins and maybe corpus maintainers which lists all current problematic sentences and where they can decide what to do with it (delete it or keep it if there are no problems).

(*) For example there's [sentence #3122484](https://tatoeba.org/eng/sentences/show/3122484) from March 2014 and tagged September 2014. It looks like it is copied from [tolearnenglish.com](https://www.tolearnenglish.com/exercises/exercise-english-2/exercise-english-94562.php) (the internet archive's [earliest capture is from 2012](https://web.archive.org/web/20121114114734/https://www.tolearnenglish.com/exercises/exercise-english-2/exercise-english-94562.php)) and their [copyright page](https://www.tolearnenglish.com/copyright.php) is pretty clear that they don't want their content used somewhere else.

**Comment by trang** [comment]
I found the reason behind the sentences with license `null`: https://github.com/Tatoeba/tatoeba2/issues/1631#issuecomment-490354101.

These sentences were part of a set of sentences that were not restored after our crash in 2017. When I restored these sentences, I didn't set the license. So they are supposed to have a CC BY license.

Now there are still 5 sentences which were not part of the restored sentences, but had a license `null`:
```
7045608
7045609
7045610
7045611
7045612
```

I'm not sure what's the story behind these. It could have been a punctual bug. In any case, they're supposed to be CC BY as well. They are sentences from Guybrush88, one of our corpus maintainers. There's no reason why we would set the license to `null`.

**Comment by trang** [comment]
> Suppose we implement the changes. What will happen afterwards?

What happens afterwards is that we should have less confusion regarding sentences in red.

[Ooneykcall's reaction at seeing Tom_Facts' sentences marked in red](https://tatoeba.org/eng/wall/show_message/34630#message_34630) was that it's unfair because (I quote) _they are grammatically correct and understandable, regardless of not making sense in the real world_. So there is an instinctive association between red and linguistically incorrect.

In my mind, we could just display these sentences as normal sentences and exclude them from our released corpus. This way there is no more confusion. 
- Sentence is red = linguistically incorrect
- Sentence not red = linguistically correct (regardless of copyright or any other issue)

However, @agrodet had a fair point when he mentioned that possible copyright infringements should have a visual indicator to distinguish them from normal sentence. I'm not entirely convinced we should, but that's something to be considered.

> So I agree with CK and I think the sentence shouldn't be displayed at all anywhere on tatoeba.org except perhaps a special page for admins and maybe corpus maintainers which lists all current problematic sentences and where they can decide what to do with it (delete it or keep it if there are no problems).

My concern, if we do that, is that things become less transparent. I think that figuring out what to do with possible copyright infringement is a conversation that every member of Tatoeba should be allowed to be part of. It's not a choice that should be left to admins and corpus maintainers alone.

I could give up on transparency if having the content publicly exposed is somehow significantly detrimental to whoever owns the intellectual property. But that's rarely, if ever, the case. In most cases, there is almost no harm because we stop contributors before it gets over the top. We benefit from the fact that we are a small community and Tatoeba isn't so widely used.

The general scenario that I imagine is something like this:
- Once in a while, someone will contribute sentences which are challenging our standards in terms of respect of intellectual property.
- When we notice that, we block the contributor from adding more sentences and we remove the CC BY license from their sentence.
  - Deleting right away the sentences as soon as we have the slightest doubt would be too harsh and would not be a good experience for the contributor if we happened to be wrong.
  - However, leaving the sentences in the CC BY corpus would be damaging if we happened to be right.
- We send a private message to notify the contributor of the situation. 
- The contributor and the rest of the community can bring arguments on why we should restore the CC BY license, or on why we should completely remove the sentences. The discussion(s) can take place in private messages, in the comments on the sentences, or on the Wall.
- Based on the discussions, we eventually either delete the sentences, or restore the CC BY license.

I understand the worry of having potentially copyrighted content being visible on the Tatoeba website, but when we only have a small sample of the overall work, then we should be able to fallback to fair use or (some similar concept). I mean, once we have acknowledged that the sample is problematic, it becomes something that we are researching, as we try to answer the question "Is it okay to include these sentences in the Tatoeba corpus?".

**Comment by AndiPersti** [comment]
> I think `for Tatoeba only` is a bit misleading though, because it feels like the contributor's wish was explicitly that Tatoeba does not share their audio, whereas in fact, we don't know what is their wish.
> 
> And to be fair, for non CC BY compatible sentences we technically don't know the original's author wish either. So I'd be tempted to try and come up with a wording that convey this idea for both textual and audio contributions.

Just brainstorming: `License unknown`, `License unclear`, `License uncertain`, `License needs to be checked`, `License not known`, `License questionable`, `License problematic`, `License unsettled`, `License in question`

> Allow admins to change the license from CC-BY to null.

I've just noticed that setting the license to `null` from the UI isn't possible with the current version because this leads to a database constraint violation in the `contributions` table when the license change will be logged. The `text` column of that table doesn't allow `null` values: https://github.com/Tatoeba/tatoeba2/blob/fa083a93f7eb26620cd6a92a317270517e2a73b4/docs/database/tables/contributions.sql#L38

In my first version I've used the empty string as it is done with the audio licenses.

**Comment by AndiPersti** [comment]
> My concern, if we do that, is that things become less transparent. I think that figuring out what to do with possible copyright infringement is a conversation that every member of Tatoeba should be allowed to be part of. It's not a choice that should be left to admins and corpus maintainers alone.

In my experience discussing specific copyright issues (i.e. copyright status of a specific sentence) with random people from the internet (myself included) end in talk because at the end only a lawyer can tell you what is a copyright infringement. But your mileage may vary :-)

> but when we only have a small sample of the overall work, then we should be able to fallback to fair use or (some similar concept).

Just a sidenote: As I understand it French law (and in general European law) doesn't have a smiliar concept like the fair use doctrine in the US. And the few [exceptions](https://fr.wikipedia.org/wiki/Droit_d%27auteur_en_France#Les_exceptions_au_droit_d%E2%80%99auteur) are rather restricted and mostly require proper acknowledgement of the source.

**Comment by ckjpn** [comment]
> Re: I think for Tatoeba only is a bit misleading though, because it feels like the contributor's wish was explicitly that Tatoeba does not share their audio, ...

I've been using "no license fo offside use" in the audio list names since before tatoeba.org had a way to specify licenses.

https://tatoeba.org/eng/sentences_lists/of_user/CK/audio%20-/page:1/sort:modified/direction:desc

This following page hasn't been updated in a while, but I primarily started naming audio lists with fields delimited by " - " to create the table on this page.

http://aitech.ac.jp/~iteslj/a4esl/temporary/tatoeba/lists/audio.html

**Comment by trang** [comment]
On the technical side, this is solved. The remaining is manual work. Closing.

**Comment by DJ-Saidez** [comment]
I suggested an "@unblock" tag that non-admins could use to distinguish these OK sentences for admins to open back up. This is mostly for use when a user's sentences are mass-blocked, but it's not a copyright issue.

It'd be nice if there was an easier way to tell, in some cases, why a sentence, a group of sentences, or a user's entire contributions were blocked, so it can be referenced in the future.
For example: https://tatoeba.org/en/user/profile/new_world_order
There aren't any comments or anything that explain why the account was blocked, and the sentences are fine to me.

**Comment by RyckRichards** [comment]
Maybe we can open another ticket for that, but it would be good if
AC/CMs/Admins know why a certain user has been blocked by displaying
something on his/her profile.

**Comment by trang** [comment]
@DJ-Saidez @RyckRichards Feel free to open a new ticket about having more information on why a sentence has been marked as red.
